src/Workleap.Extensions.Mongo.Ephemeral/EphemeralMongoBuilderExtensions.cs

```diff
@@ -23,4 +24,29 @@ public static MongoBuilder UseEphemeralRealServer(this MongoBuilder builder)
 
         return builder;
     }
+
+    public static MongoBuilder UseEphemeralRealServer(this MongoBuilder builder, string connectionString)
+    {
+        builder.Services.AddSingleton<DefaultDatabaseNameHolder>();
+        builder.Services.AddSingleton<IMongoClientProvider>(serviceProvider =>
+        {
+            var client = new MongoClient(connectionString);
+            var defaultDatabaseNameHolder = serviceProvider.GetRequiredService<DefaultDatabaseNameHolder>();
+            return new DisposableMongoClientProvider(new SingletonMongoClientProvider(client), defaultDatabaseNameHolder);
+        });
+
+        return builder;
+    }
```

Comment:
I think we could simplify this by passing the new MongoClient(connectionString) to the overload below